Array comparisons do not work

I cannot get the new V2.x API to work when testing against a JSON array. It's quite possible I'm doing something wrong, but I've exhausted everything I can think of. 

However, I am able to get the array test to work by using the Jasmine `expect` function.

Here's a script that runs two specs, using a public "minor planets" RESTful API. The first spec uses the builtin Frisby `expect`, the second the Jasmine `expect` function.

```javascript
const frisby = require('frisby');

const expected_value = [
    {
        rms: 0.6,
        epoch: 'K1794',
        readable_des: '(1) Ceres',
        H: 3.34,
        num_obs: 6672,
        ref: 'MPO399990',
        G: 0.12,
        last_obs: '20170305',
        comp: 'MPCLINUX  ',
        M: 309.49412,
        U: ' ',
        e: 0.0756074,
        a: 2.7674094,
        om: 80.30888,
        pert_p: '30h',
        d: 0.21408881,
        i: 10.59322,
        des: '00001',
        flags: '0000',
        num_opp: 113,
        w: 73.02368,
        pert_c: 'M-v'
    },
    {
        rms: 0.58,
        epoch: 'K1794',
        readable_des: '(2) Pallas',
        H: 4.13,
        num_obs: 7877,
        ref: 'MPO410073',
        G: 0.11,
        last_obs: '20170601',
        comp: 'MPCLINUX  ',
        M: 291.65136,
        U: ' ',
        e: 0.2305974,
        a: 2.7730852,
        om: 173.08718,
        pert_p: '28h',
        d: 0.21343186,
        i: 34.83792,
        des: '00002',
        flags: '0000',
        num_opp: 108,
        w: 309.99154,
        pert_c: 'M-v'
    },
    {
        rms: 0.59,
        epoch: 'K1794',
        readable_des: '(3) Juno',
        H: 5.33,
        num_obs: 6863,
        ref: 'MPO410073',
        G: 0.32,
        last_obs: '20170521',
        comp: 'MPCLINUX  ',
        M: 259.23491,
        U: ' ',
        e: 0.2568534,
        a: 2.6685312,
        om: 169.85828,
        pert_p: '38h',
        d: 0.22609744,
        i: 12.98996,
        des: '00003',
        flags: '0000',
        num_opp: 104,
        w: 248.20648,
        pert_c: 'M-v'
    }
];

// First run a spec using the built in Frisby 'expect'. This fails.
it("should work for a 'JSON' field", function (doneFn) {
    frisby
        .get('http://www.asterank.com/api/mpc?limit=3')
        .expect('status', 200)
        .expect('json', '', expected_value)
        .done(doneFn);
});

// Now run a spec using Jasmine's expect. This works
it("should work for a Jasmine 'expect' clause", function (doneFn) {
    frisby
        .get('http://www.asterank.com/api/mpc?limit=3')
        .expect('status', 200)
        .then(function (res) {
            expect(res.body).toEqual(expected_value);
        })
        .done(doneFn);
});
```

Here's the output from the script:

```shell
$ jasmine spec/server/array_spec.js
Started
F.

Failures:
1) should work for a 'JSON' field
  Message:
    Failed: Response [ [{"rms":0.6,"epoch":"K1794","readable_des":"(1) Ceres","H":3.34,"num_obs":6672,"ref":"MPO399990","G":0.12,"last_obs":"20170305","comp":"MPCLINUX  ","M":309.49412,"U":" ","e":0.0756074,"a":2.7674094,"om":80.30888,"pert_p":"30h","d":0.21408881,"i":10.59322,"des":"00001","flags":"0000","num_opp":113,"w":73.02368,"pert_c":"M-v"},{"rms":0.58,"epoch":"K1794","readable_des":"(2) Pallas","H":4.13,"num_obs":7877,"ref":"MPO410073","G":0.11,"last_obs":"20170601","comp":"MPCLINUX  ","M":291.65136,"U":" ","e":0.2305974,"a":2.7730852,"om":173.08718,"pert_p":"28h","d":0.21343186,"i":34.83792,"des":"00002","flags":"0000","num_opp":108,"w":309.99154,"pert_c":"M-v"},{"rms":0.59,"epoch":"K1794","readable_des":"(3) Juno","H":5.33,"num_obs":6863,"ref":"MPO410073","G":0.32,"last_obs":"20170521","comp":"MPCLINUX  ","M":259.23491,"U":" ","e":0.2568534,"a":2.6685312,"om":169.85828,"pert_p":"38h","d":0.22609744,"i":12.98996,"des":"00003","flags":"0000","num_opp":104,"w":248.20648,"pert_c":"M-v"}] ] does not contain provided JSON [ [{"rms":0.6,"epoch":"K1794","readable_des":"(1) Ceres","H":3.34,"num_obs":6672,"ref":"MPO399990","G":0.12,"last_obs":"20170305","comp":"MPCLINUX  ","M":309.49412,"U":" ","e":0.0756074,"a":2.7674094,"om":80.30888,"pert_p":"30h","d":0.21408881,"i":10.59322,"des":"00001","flags":"0000","num_opp":113,"w":73.02368,"pert_c":"M-v"},{"rms":0.58,"epoch":"K1794","readable_des":"(2) Pallas","H":4.13,"num_obs":7877,"ref":"MPO410073","G":0.11,"last_obs":"20170601","comp":"MPCLINUX  ","M":291.65136,"U":" ","e":0.2305974,"a":2.7730852,"om":173.08718,"pert_p":"28h","d":0.21343186,"i":34.83792,"des":"00002","flags":"0000","num_opp":108,"w":309.99154,"pert_c":"M-v"},{"rms":0.59,"epoch":"K1794","readable_des":"(3) Juno","H":5.33,"num_obs":6863,"ref":"MPO410073","G":0.32,"last_obs":"20170521","comp":"MPCLINUX  ","M":259.23491,"U":" ","e":0.2568534,"a":2.6685312,"om":169.85828,"pert_p":"38h","d":0.22609744,"i":12.98996,"des":"00003","flags":"0000","num_opp":104,"w":248.20648,"pert_c":"M-v"}] ]
  Stack:
    AssertionError: Response [ [{"rms":0.6,"epoch":"K1794","readable_des":"(1) Ceres","H":3.34,"num_obs":6672,"ref":"MPO399990","G":0.12,"last_obs":"20170305","comp":"MPCLINUX  ","M":309.49412,"U":" ","e":0.0756074,"a":2.7674094,"om":80.30888,"pert_p":"30h","d":0.21408881,"i":10.59322,"des":"00001","flags":"0000","num_opp":113,"w":73.02368,"pert_c":"M-v"},{"rms":0.58,"epoch":"K1794","readable_des":"(2) Pallas","H":4.13,"num_obs":7877,"ref":"MPO410073","G":0.11,"last_obs":"20170601","comp":"MPCLINUX  ","M":291.65136,"U":" ","e":0.2305974,"a":2.7730852,"om":173.08718,"pert_p":"28h","d":0.21343186,"i":34.83792,"des":"00002","flags":"0000","num_opp":108,"w":309.99154,"pert_c":"M-v"},{"rms":0.59,"epoch":"K1794","readable_des":"(3) Juno","H":5.33,"num_obs":6863,"ref":"MPO410073","G":0.32,"last_obs":"20170521","comp":"MPCLINUX  ","M":259.23491,"U":" ","e":0.2568534,"a":2.6685312,"om":169.85828,"pert_p":"38h","d":0.22609744,"i":12.98996,"des":"00003","flags":"0000","num_opp":104,"w":248.20648,"pert_c":"M-v"}] ] does not contain provided JSON [ [{"rms":0.6,"epoch":"K1794","readable_des":"(1) Ceres","H":3.34,"num_obs":6672,"ref":"MPO399990","G":0.12,"last_obs":"20170305","comp":"MPCLINUX  ","M":309.49412,"U":" ","e":0.0756074,"a":2.7674094,"om":80.30888,"pert_p":"30h","d":0.21408881,"i":10.59322,"des":"00001","flags":"0000","num_opp":113,"w":73.02368,"pert_c":"M-v"},{"rms":0.58,"epoch":"K1794","readable_des":"(2) Pallas","H":4.13,"num_obs":7877,"ref":"MPO410073","G":0.11,"last_obs":"20170601","comp":"MPCLINUX  ","M":291.65136,"U":" ","e":0.2305974,"a":2.7730852,"om":173.08718,"pert_p":"28h","d":0.21343186,"i":34.83792,"des":"00002","flags":"0000","num_opp":108,"w":309.99154,"pert_c":"M-v"},{"rms":0.59,"epoch":"K1794","readable_des":"(3) Juno","H":5.33,"num_obs":6863,"ref":"MPO410073","G":0.32,"last_obs":"20170521","comp":"MPCLINUX  ","M":259.23491,"U":" ","e":0.2568534,"a":2.6685312,"om":169.85828,"pert_p":"38h","d":0.22609744,"i":12.98996,"des":"00003","flags":"0000","num_opp":104,"w":248.20648,"pert_c":"M-v"}] ]
        at jsonContainsAssertion (/home/tkeffer/git/tkeffer/weert-js/node_modules/frisby/src/frisby/expects.js:79:14)
        at Object.withPath (/home/tkeffer/git/tkeffer/weert-js/node_modules/frisby/src/frisby/utils.js:11:12)
        at FrisbySpec.json (/home/tkeffer/git/tkeffer/weert-js/node_modules/frisby/src/frisby/expects.js:70:11)
        at FrisbySpec._addExpect.e (/home/tkeffer/git/tkeffer/weert-js/node_modules/frisby/src/frisby/spec.js:408:23)
        at then (/home/tkeffer/git/tkeffer/weert-js/node_modules/frisby/src/frisby/spec.js:299:26)
        at _fetch._fetch.then (/home/tkeffer/git/tkeffer/weert-js/node_modules/frisby/src/frisby/spec.js:231:18)
        at process._tickCallback (internal/process/next_tick.js:103:7)

2 specs, 1 failure
Finished in 0.742 seconds
```
Again, it's entirely possible I'm doing something wrong. In particular, perhaps the semantics of the `path` argument has changed?

That was the very first thing I tried. Didn't work. Here's the output:

```shell
Started
.F.

Failures:
1) should work for a 'JSON' field
  Message:
    Failed: Response [ [{"rms":0.6,"epoch":"K1794","readable_des":"(1) Ceres","H":3.34,"num_obs":6672,"ref":"MPO399990","G":0.12,"last_obs":"20170305","comp":"MPCLINUX  ","M":309.49412,"U":" ","e":0.0756074,"a":2.7674094,"om":80.30888,"pert_p":"30h","d":0.21408881,"i":10.59322,"des":"00001","flags":"0000","num_opp":113,"w":73.02368,"pert_c":"M-v"},{"rms":0.58,"epoch":"K1794","readable_des":"(2) Pallas","H":4.13,"num_obs":7877,"ref":"MPO410073","G":0.11,"last_obs":"20170601","comp":"MPCLINUX  ","M":291.65136,"U":" ","e":0.2305974,"a":2.7730852,"om":173.08718,"pert_p":"28h","d":0.21343186,"i":34.83792,"des":"00002","flags":"0000","num_opp":108,"w":309.99154,"pert_c":"M-v"},{"rms":0.59,"epoch":"K1794","readable_des":"(3) Juno","H":5.33,"num_obs":6863,"ref":"MPO410073","G":0.32,"last_obs":"20170521","comp":"MPCLINUX  ","M":259.23491,"U":" ","e":0.2568534,"a":2.6685312,"om":169.85828,"pert_p":"38h","d":0.22609744,"i":12.98996,"des":"00003","flags":"0000","num_opp":104,"w":248.20648,"pert_c":"M-v"}] ] does not contain provided JSON [ [{"rms":0.6,"epoch":"K1794","readable_des":"(1) Ceres","H":3.34,"num_obs":6672,"ref":"MPO399990","G":0.12,"last_obs":"20170305","comp":"MPCLINUX  ","M":309.49412,"U":" ","e":0.0756074,"a":2.7674094,"om":80.30888,"pert_p":"30h","d":0.21408881,"i":10.59322,"des":"00001","flags":"0000","num_opp":113,"w":73.02368,"pert_c":"M-v"},{"rms":0.58,"epoch":"K1794","readable_des":"(2) Pallas","H":4.13,"num_obs":7877,"ref":"MPO410073","G":0.11,"last_obs":"20170601","comp":"MPCLINUX  ","M":291.65136,"U":" ","e":0.2305974,"a":2.7730852,"om":173.08718,"pert_p":"28h","d":0.21343186,"i":34.83792,"des":"00002","flags":"0000","num_opp":108,"w":309.99154,"pert_c":"M-v"},{"rms":0.59,"epoch":"K1794","readable_des":"(3) Juno","H":5.33,"num_obs":6863,"ref":"MPO410073","G":0.32,"last_obs":"20170521","comp":"MPCLINUX  ","M":259.23491,"U":" ","e":0.2568534,"a":2.6685312,"om":169.85828,"pert_p":"38h","d":0.22609744,"i":12.98996,"des":"00003","flags":"0000","num_opp":104,"w":248.20648,"pert_c":"M-v"}] ]
  Stack:
    AssertionError: Response [ [{"rms":0.6,"epoch":"K1794","readable_des":"(1) Ceres","H":3.34,"num_obs":6672,"ref":"MPO399990","G":0.12,"last_obs":"20170305","comp":"MPCLINUX  ","M":309.49412,"U":" ","e":0.0756074,"a":2.7674094,"om":80.30888,"pert_p":"30h","d":0.21408881,"i":10.59322,"des":"00001","flags":"0000","num_opp":113,"w":73.02368,"pert_c":"M-v"},{"rms":0.58,"epoch":"K1794","readable_des":"(2) Pallas","H":4.13,"num_obs":7877,"ref":"MPO410073","G":0.11,"last_obs":"20170601","comp":"MPCLINUX  ","M":291.65136,"U":" ","e":0.2305974,"a":2.7730852,"om":173.08718,"pert_p":"28h","d":0.21343186,"i":34.83792,"des":"00002","flags":"0000","num_opp":108,"w":309.99154,"pert_c":"M-v"},{"rms":0.59,"epoch":"K1794","readable_des":"(3) Juno","H":5.33,"num_obs":6863,"ref":"MPO410073","G":0.32,"last_obs":"20170521","comp":"MPCLINUX  ","M":259.23491,"U":" ","e":0.2568534,"a":2.6685312,"om":169.85828,"pert_p":"38h","d":0.22609744,"i":12.98996,"des":"00003","flags":"0000","num_opp":104,"w":248.20648,"pert_c":"M-v"}] ] does not contain provided JSON [ [{"rms":0.6,"epoch":"K1794","readable_des":"(1) Ceres","H":3.34,"num_obs":6672,"ref":"MPO399990","G":0.12,"last_obs":"20170305","comp":"MPCLINUX  ","M":309.49412,"U":" ","e":0.0756074,"a":2.7674094,"om":80.30888,"pert_p":"30h","d":0.21408881,"i":10.59322,"des":"00001","flags":"0000","num_opp":113,"w":73.02368,"pert_c":"M-v"},{"rms":0.58,"epoch":"K1794","readable_des":"(2) Pallas","H":4.13,"num_obs":7877,"ref":"MPO410073","G":0.11,"last_obs":"20170601","comp":"MPCLINUX  ","M":291.65136,"U":" ","e":0.2305974,"a":2.7730852,"om":173.08718,"pert_p":"28h","d":0.21343186,"i":34.83792,"des":"00002","flags":"0000","num_opp":108,"w":309.99154,"pert_c":"M-v"},{"rms":0.59,"epoch":"K1794","readable_des":"(3) Juno","H":5.33,"num_obs":6863,"ref":"MPO410073","G":0.32,"last_obs":"20170521","comp":"MPCLINUX  ","M":259.23491,"U":" ","e":0.2568534,"a":2.6685312,"om":169.85828,"pert_p":"38h","d":0.22609744,"i":12.98996,"des":"00003","flags":"0000","num_opp":104,"w":248.20648,"pert_c":"M-v"}] ]
        at jsonContainsAssertion (/home/tkeffer/git/tkeffer/weert-js/node_modules/frisby/src/frisby/expects.js:79:14)
        at Object.withPath (/home/tkeffer/git/tkeffer/weert-js/node_modules/frisby/src/frisby/utils.js:11:12)
        at FrisbySpec.json (/home/tkeffer/git/tkeffer/weert-js/node_modules/frisby/src/frisby/expects.js:70:11)
        at FrisbySpec._addExpect.e (/home/tkeffer/git/tkeffer/weert-js/node_modules/frisby/src/frisby/spec.js:408:23)
        at then (/home/tkeffer/git/tkeffer/weert-js/node_modules/frisby/src/frisby/spec.js:299:26)
        at _fetch._fetch.then (/home/tkeffer/git/tkeffer/weert-js/node_modules/frisby/src/frisby/spec.js:231:18)
        at process._tickCallback (internal/process/next_tick.js:103:7)

3 specs, 1 failure
Finished in 2.003 seconds
```

Give the script a try yourself. It's standalone. It requires only Jasmine (and Frisby)

--- a/__tests__/expects_json_spec.js
+++ b/__tests__/expects_json_spec.js
@@ -110,6 +110,62 @@ describe('expect(\'json\')', function() {
       .done(doneFn);
   });
 
+  it('should match array using json', function(doneFn) {
+    frisby.fromJSON(['a', 1, true, null])
+      .expect('json', ['a', 1, true, null])
+      .then(function() {
+        return frisby.fromJSON(['a', 1, true, null])
+          .expect('json', ['a', 1, true])
+      })
+      .then(function() {
+        return frisby.fromJSON(['a', 1, true, null])
+          .expect('json', [1, null])
+      })
+      .then(function() {
+        return frisby.fromJSON([{a: 0}, {b: 1}, {c: 2}, [0, 1, 2]])
+          .expect('json', [{a: 0}, {b: 1}, {c: 2}, [0, 1, 2]])
+      })
+      .then(function() {
+        return frisby.fromJSON([{a: 0}, {b: 1}, {c: 2}, [0, 1, 2]])
+          .expect('json', [{a: 0}, {b: 1}, {c: 2}])
+      })
+      .then(function() {
+        return frisby.fromJSON([{a: 0}, {b: 1}, {c: 2}, [0, 1, 2]])
+          .expect('json', [{b: 1}, [0, 1, 2]])
+      })
+      .done(doneFn);
+  });
+
+  it('should error different array using json', function(doneFn) {
+    frisby.fromJSON(['a', 1, true, null])
+      .expectNot('json', ['a', 0, true, null])
+      .then(function() {
+        return frisby.fromJSON(['a', 1, true, null])
+          .expectNot('json', ['a', 1, true, null, false])
+      })
+      .then(function() {
+        return frisby.fromJSON(['a', 1, true, null])
+          .expectNot('json', [0, null])
+      })
+      .then(function() {
+        return frisby.fromJSON([{a: 0}, {b: 1}, {c: 2}, [0, 1, 2]])
+          .expectNot('json', [{a: 0}, {b: 1}, {c: 1}, [0, 1, 2]])
+      })
+      .then(function() {
+        return frisby.fromJSON([{a: 0}, {b: 1}, {c: 2}, [0, 1, 2]])
+          .expectNot('json', [{a: 0}, {b: 1}, {c: 1}])
+      })
+      .then(function() {
+        return frisby.fromJSON([{a: 0}, {b: 1}, {c: 2}, [0, 1, 2]])
+          .expectNot('json', [{b: 1}, [0, 1, 1]])
+      })
+      .then(function() {
+        return frisby.fromJSON({a: 0})
+          .expectNot('json', [{a: 0}])
+      })
+      .done(doneFn);
+  });
+
 });
 
 describe('expect(\'jsonStrict\')', function() {

--- a/src/frisby/expects.js
+++ b/src/frisby/expects.js
@@ -68,10 +68,13 @@ const expects = {
     incrementAssertionCount();
 
     utils.withPath(path, response._body, function jsonContainsAssertion(jsonChunk) {
-      let failMsg = "Response [ " + JSON.stringify(jsonChunk) + " ] does not contain provided JSON [ " + JSON.stringify(json) + " ]";
+      let failMsg = `Response [ ${JSON.stringify(jsonChunk)} ] does not contain provided JSON [ ${JSON.stringify(json)} ]`;
 
-      if (_.isObject(json)) {
-        // Object/aray test
+      if (_.isArray(json)) {
+        // Aaray test
+        assert.ok(_.differenceWith(json, jsonChunk, _.isEqual).length === 0, failMsg);
+      } else if (_.isObject(json)) {
+        // Object test
         assert.ok(_.some([jsonChunk], json), failMsg);
       } else {
         // Single value test

